### DISCUSSION TITLE
StaEDIXMLStreamWriter isn't validating against provided schema

### DISCUSSION THREAD

**Discussion by d-floyd** [xlate/staedi#282] [Q&A] [answered]
I have validation working perfectly for reading an x12 payload (the custom report hook is awesome). However,  when I try to use the same schema when _writing_ , validation doesn't seem to be occurring.
I've provided test code/sample schema information. 
I configure the writer with the schema set, and generate the x12 payload (I expected it to fail)
if I take the payload that was generated , and try to read it -- it fails as expected.
Is my writer configuration not correct? or does the validation not work on the XMLStreamWriter?
[OutputValidation.zip](https://github.com/xlate/staedi/files/9124848/OutputValidation.zip)

As always, thanks for the help.

**Comment by MikeEdgar** [comment, marked as answer]
@d-floyd - you will also need to set the control schema on the `EDIStreamWriter`. You can obtain one of the built-in control schemas using `SchemaFactory#getControlSchema`. For your example it would be `getControlSchema("X12", new String[] { "00501" });`. 

The reader automatically loads them, but not (yet) for the writer. I opened https://github.com/xlate/staedi/issues/283 for that enhancement.

**Reply by d-floyd** [reply]
As usual. Amazing response time to questions asked -- you're top notch.
Just a clarification for anyone looking at this answer above. 
You must set the control schema, before you set the transaction schema. 
If you don't you'll get SEGMENT_NOT_IN_DEFINED_TRANSACTION_SET for the SE element.
